### PR TITLE
Fix deadlock in zlog flush deinit

### PR DESCRIPTION
**Pull request by jw-msft**

### CODE REVIEW

**Review by nihemstr — APPROVED** [review]

File: src/logging/zlog/src/zlog.c [hunk 1]
```diff
@@ -48,16 +48,12 @@ static const char level_names[] = { 'D', 'I', 'W', 'E' }; // Must align with ZLO
 static FILE* zlog_fout = NULL;
 static char* zlog_file_log_dir = NULL;
 static char* zlog_file_log_prefix = NULL;
+static time_t zlog_last_flushed;
```
Comment (JeffMill) [inline src/logging/zlog/src/zlog.c:51]: init this.

Comment (JeffMill) [reply src/logging/zlog/src/zlog.c:51]: looks like zlog_init does, but doesn't hurt to set to 0.

File: src/logging/zlog/src/zlog.c [hunk 2]
```diff
@@ -375,81 +368,14 @@ void zlog_log(enum ZLOG_SEVERITY msg_level, const char* func, const char* fmt, .
...
-{
-    zlog_buffer_flush_thread_run = true;
-    if (pthread_create(&_zlog_flush_thread, NULL, zlog_buffer_flush_thread, NULL) == 0)
+    if (msg_level == ZLOG_ERROR || (seconds - zlog_last_flushed) >= ZLOG_FLUSH_INTERVAL_SEC)
     {
-        _is_flush_thread_initialized  = true;
-    }
-}
+        _zlog_buffer_lock();
+        _zlog_flush_buffer();
```
Comment (JeffMill) [inline src/logging/zlog/src/zlog.c:374]: zlog_flush_buffer should do the lock and unlock, no?

**Review by JeffMill — CHANGES_REQUESTED** [review]

File: src/logging/zlog/src/zlog.c (hunk 2 above)
Comment (jw-msft) [reply src/logging/zlog/src/zlog.c:374]: line 353 where it rolls the log seems to want to hold the lock for longer so I think that's why _zlog_flush_buffer() has a comment that says to acquire the lock before calling.

**Review by JeffMill — APPROVED** [review]

**Pull request merged**